File: clients/admin/src/mutations/AddUser.ts
import client from '../modules/apollo';
import mutationPromise from '../helpers/mutationPromise';
import gql from 'graphql-tag';

const mutation = gql`
  mutation AddUserMutation($input: AddUserInput!) {
    addUser(input: $input) {
      success
      message
      userErrors {
        field
        message
      }
    }
  }
`;

function commit(variables: AddUserMutationVariables) {
  return mutationPromise((resolve, reject) => {
    client
      .mutate({
        mutation,
        variables
      })
      .then(({ data: { addUser } }) => {
        resolve(addUser as MutationResult);
      })
      .catch(error => {
        reject(error);
      });
  });
}

export default { commit };

File: app/graphql/mutations/add_user.rb
class Mutations::AddUser < Mutations::BaseMutation
  graphql_name "AddUser"

  argument :input, Inputs::AddUserInput, required: true

  field :user, Types::User, null: false
  field :success, Boolean, null: false
  field :message, String, null: true
  field :userErrors, [Types::Error], null: true
end
